lib/resty/etcd/v3.lua

```diff
@@ -541,6 +542,15 @@ local function request_chunk(self, method, host, port, path, opts, timeout)
 
         return body
     end
+
+    if opts.need_cancel == true then
+        return {
+            func = read_watch,
+            http_cli = http_cli
+        }
+    else
+        return read_watch
+    end
```

Comment:
@nic-chen is there any needs to also change the return structure of this place

Reply:
yes，good